### PR TITLE
Update packaging guidelines cont

### PR DESCRIPTION
**Pull request by tmds**
This addresses comments https://github.com/dotnet/docs/pull/13196#issuecomment-528585112 and below.

@omajid @dagood ptal

cc @leecow

### CODE REVIEW

File: docs/core/build/distribution-packaging.md
Comment (dagood) [inline docs/core/build/distribution-packaging.md]: I'd be curious to learn whether it's generally good to rely on transitive dependencies or not. I don't have an objection, though.

File: docs/core/build/distribution-packaging.md
Comment (dagood) [inline docs/core/build/distribution-packaging.md]: Should we have a note<sup>maybe a `sup`</sup> about what this means for anyone reading the doc for a distro that doesn't do directory ownership? (Only Fedora-like distros mentioned in the earlier PR.)

Might also be useful to call out `dotnet_root`, in the file section it looks like `.` means the same thing. Or is it a true identifier, and changing where `dotnet-host` is installed changes the rest?

**Review by dagood — APPROVED** [review]
Seems reasonable, a couple suggestions.

File: docs/core/build/distribution-packaging.md
Comment (tmds) [reply docs/core/build/distribution-packaging.md]: @omajid ?

File: docs/core/build/distribution-packaging.md
Comment (tmds) [reply docs/core/build/distribution-packaging.md]: > Only Fedora-like distros mentioned in the earlier PR.

@omajid is the concept of directory ownership rpm specitic? Do you know if it applies to deb?

> Might also be useful to call out dotnet_root, in the file section it looks like . means the same thing. Or is it a true identifier, and changing where dotnet-host is installed changes the rest?

I've addressed this by consistently using `{dotnet_root}`

File: docs/core/build/distribution-packaging.md
Comment (tmds) [inline docs/core/build/distribution-packaging.md]: I don't have a strong preference, but as an alternative to the 

> **Dependencies:** `dotnet-host` (provides `{dotnet_root}`)

We could add `{dotnet_root}` to the **Provides shared directory:**.

@omajid @dagood do you have some thoughts about this?

File: docs/core/build/distribution-packaging.md
Comment (leecow) [inline docs/core/build/distribution-packaging.md]: I may be misunderstanding something At the bottom of this doc you mention the following, which seems to contradict the "Provides" section of `aspnetcore-runtime-[major].[minor]`.

> - The third-party libraries under `shared/Microsoft.AspNetCore.All` can't be easily built from source. So that folder is omitted from the `aspnetcore-runtime` package.

Comment (tmds) [reply docs/core/build/distribution-packaging.md]: I need to clarify in the doc what _Provides shared directory_ means. It's about the folder itself, and not the content under it. _Contains_ is about the content.

All folders listed as _Provides shared directory_ are folders which may be used by different packages. For example, aspnetcore-runtime-2.1 and aspnetcore-runtime-2.2 may both create the `{dotnet_root}/shared/Microsoft.AspNetCore.App` under which they place their own versioned folder.

File: docs/core/build/distribution-packaging.md
Comment (omajid) [reply docs/core/build/distribution-packaging.md]: My 2 cents: Maybe this is too much detail? I am not sure if something like "provides shared directory" needs to be completely specified.

File: docs/core/build/distribution-packaging.md
Comment (omajid) [reply docs/core/build/distribution-packaging.md]: I don't think it's an exact science. If it works, it works :smile: 

It seems to me like the SDK depends on `shared/Microsoft.NETCore.App` itself, so a dependency on the runtime package makes sense. But it doesn't need to be mandatory.

File: docs/core/build/distribution-packaging.md
Comment (omajid) [reply docs/core/build/distribution-packaging.md]: I would leave it out, given that it doesn't apply to all packaging systems.

Packagers are free to deviate from this doc when it makes sense, right?

File: docs/core/build/distribution-packaging.md
Comment (omajid) [inline docs/core/build/distribution-packaging.md]: I just noticed that this line says `netstandard-targeting-pack-[netstandard_major].[netstandard_minor]` but the list of packages below says `netstandard-targeting-pack-[major].[minor]`. That seems like a mismatch. It should be the netstandard version in both cases, right?

Comment (tmds) [reply docs/core/build/distribution-packaging.md]: I made it explicit here to make clear this is not matching the `[major].[minor]` of the package. I'll use it everywhere so it's less confusing.

**Comment by tmds** [comment]
All review comments addressed. @dagood @omajid can you approve the PR if it is good for merging?

File: docs/core/build/distribution-packaging.md
Comment (dagood) [inline docs/core/build/distribution-packaging.md]: It would be nice to have this described somewhere... but this note LGTM--hard to make it clear to readers not already familiar with this requirement without putting a lot more info in this doc and probably cluttering it up. Selfishly, I'd like to get the Microsoft RPMs up to snuff, and know what the consequences are of getting it wrong.

**Review by dagood — APPROVED** [review]
One minor suggestion (section move) and one comment, LGTM either way though.

**Review by omajid — APPROVED** [review]
LGTM. Thanks!

File: docs/core/build/distribution-packaging.md
Comment (tmds) [reply docs/core/build/distribution-packaging.md]: moving this up

Comment (tmds) [reply docs/core/build/distribution-packaging.md]: Yes, we can assume that the package maintainer knows what this is about.

**Comment by tmds** [comment]
@mairaw, based on @dagood and @omajid review, this is good for merging.

@omajid @dagood thanks for all your feedback to get this together!

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [inline docs/core/build/distribution-packaging.md]:
```suggestion
The folders marked with `(*)` are used by multiple packages. Some package formats (for example, `rpm`) require special handling of such folders. The package maintainer must take care of this.
```

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [inline docs/core/build/distribution-packaging.md]: can you also update ms.date (line 5) on this file?

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [inline docs/core/build/distribution-packaging.md]:
```suggestion
- (16) **/etc/dotnet/install_location** is a file that contains the full path for `{dotnet_root}`. The path may end with a newline. It's not necessary to add this file when the root is `/usr/share/dotnet`.
```

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [inline docs/core/build/distribution-packaging.md]:
```suggestion
Dependencies between packages should use an _equal or greater than_ version requirement. For example, `dotnet-sdk-2.2:2.2.401` requires `aspnetcore-runtime-2.2 >= 2.2.6`. This makes it possible for the user to upgrade their installation via a root package (for example, `dnf update dotnet-sdk-2.2`).
```

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [inline docs/core/build/distribution-packaging.md]:
```suggestion
The `dotnet-runtime-deps-[major].[minor]` requires understanding the _distro-specific dependencies_. Because the distro build system may be able to derive this automatically, the package is optional, in which case these dependencies are added directly to the `dotnet-runtime-[major].[minor]` package.
```

**Review by mairaw — APPROVED** [review]
Thank you @tmds. Left a few comments, some unrelated to your changes. @leecow @jamshedd can one of you review this as well?

**Comment by leecow** [comment]
> Thank you @tmds. Left a few comments, some unrelated to your changes. @leecow @jamshedd can one of you review this as well?

Suggested edits look good.

**Comment by mairaw** [comment]
I've applied the suggestions so we can get this merged. Thanks so much @tmds!

**Pull request merged**
